enginetest/query_plans.go

```diff
@@ -463,6 +463,13 @@ var PlanTests = []QueryPlanTest{
 			"         └─ IndexedTableAccess(mytable on [mytable.s])\n" +
 			"",
 	},
+	{
+		Query: `SELECT * FROM mytable WHERE i in (1, 2, 3, 4)`,
+		ExpectedPlan: "Filter(mytable.i HASH IN (1, 2, 3, 4))\n" +
+			" └─ Projected table access on [i s]\n" +
+			"     └─ IndexedTableAccess(mytable on [mytable.i])\n" +
```

Comment:
Something is weird here.

The tests above don't get `HASH IN` treatment. Why not?

The second thing is that when we push down the `IN` clause to the table as an index, we should in principle remove that predicate from the filter. But we aren't doing that.

Could be a followup change, but not having the filter at all is even better than having it fast, and should work for IN expressions.

Reply:
the other `IN`s are converted to table joins. re: filter, i wonder if my special case is sidestepping other IN optimizations, which would be bad. i just wanted some way to verify that the transform is being applied correctly. analyzer unit tests would also accomplish that

Reply:
Yep my bad expression tree traversal was missing nested tuple ins. Made an issue for the filter/table access on the same primary key https://github.com/dolthub/go-mysql-server/issues/616.